torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -264,6 +267,214 @@ class SchedulerTopologyChecker {
+  std::unordered_map<TensorView*, size_t> cg_tensor_to_at_tensor_index;
+
+  for (auto input_index : c10::irange(number_of_inputs)) {
+    if (auto input_tensor = dynamic_cast<TensorView*>(
+            complete_fusion_->inputs()[input_index])) {
+      if (input_tensor->nDims() == 0) {
+        // A 0-dim tensor input would not need vectorization
+        continue;
+      }
+      if (input_tensor->domain()
+              ->domain()[input_tensor->nDims() - 1]
+              ->isBroadcast()) {
+        // skip the tensors with innermost iterdomain broadcasted,
+        //  as we will not vectorize these.
+        continue;
+      }
+
+      // Collect strides of the input tensor
+      TORCH_INTERNAL_ASSERT(inputs[input_index].isTensor());
+      const auto& at_tensor = inputs[input_index].toTensor();
+
+      cg_tensor_to_at_tensor_index.emplace(
+          std::make_pair(input_tensor, input_index));
+
+      // Collect alignment of the input tensor
+      auto alignment_size = collectAlignmentSize(at_tensor);
+      common_alignment_size_ = std::min(alignment_size, common_alignment_size_);
+      alignment_map_[input_tensor] = alignment_size;
+    }
+  }
+
+  // Compute max vector word size for each input,
+  //  tensors with inner most broadcast already
+  //  filtered out.  common_alignment_size_ is
+  //  computed up to this point.
+  for (auto it : cg_tensor_to_at_tensor_index) {
+    vectorword_map_[it.first] = collectMaxVectorizeSize(
+        inputs[it.second].toTensor(), common_alignment_size_);
+  }
+}
+
+size_t SchedulerRuntimeInfo::collectMaxVectorizeSize(
+    const at::Tensor& tensor,
+    size_t max_vector_size_in_byte) {
+  size_t vector_size = 1;
+  size_t next_vector_size = 2;
+  bool next_size_compatible = true;
+
+  while (next_size_compatible &&
+         next_vector_size * tensor.itemsize() <= max_vector_size_in_byte) {
+    // If inner most dimension size is not divisible by new word size
+    //  then we cannot vectorize with this width. But we do not
+    //  care if all dimensions of this tensor is 1, i.e.
+    //  input is actually a un-squeezed 0-dim tensor.
+    for (size_t i = tensor.ndimension(); i > 0; i--) {
+      if (tensor.size(i - 1) != 1) {
+        if (tensor.size(tensor.ndimension() - 1) % next_vector_size != 0 ||
+            tensor.stride(tensor.ndimension() - 1) != 1) {
+          next_size_compatible = false;
+        }
+        break;
+      }
+    }
+
+    if (!next_size_compatible) {
+      break;
+    }
+
+    // If any stride is not divisible by the next word size,
+    //  we cannot vectorize with this width.
+    for (auto stride : tensor.strides()) {
+      if (stride != 1 && stride % next_vector_size != 0) {
+        next_size_compatible = false;
+        break;
+      }
+    }
+
+    if (next_size_compatible) {
+      vector_size = next_vector_size;
+      next_vector_size *= 2;
+    }
+  }
+
+  return vector_size;
+}
+
+size_t SchedulerRuntimeInfo::getVectorizableWidth(TensorView* tv) {
```

Comment:
@csarofeen 
This will be the user-facing function for the schedulers. For input it will return the value pre-computed by `collectMaxVectorizeSize` and for other intermediate or output tensors it will do something similar to what is now in PointWise scheduler.

Reply:
@csarofeen 
I was wondering what the condition the below logic is checking. I can move the check in `RuntimeInfo` as well since it is already trying to compute max vector size. 
[https://github.com/csarofeen/pytorch/blob/20_12_3_devel/torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp#L220-L226](https://github.com/csarofeen/pytorch/blob/20_12_3_devel/torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp#L220-L226)

Once I know I have all the necessary checks I can try to propagate these to pointwise schedulers in this PR if you'd like.